### PR TITLE
fix(#3842): DnD Nested Drop Regions example can't focus child drop target

### LINKED ISSUE #3842: DnD Nested Drop Regions example can't focus child drop target
# 🐛 Bug Report 
1. Open [Drag and Drop, nested drop targets story](https://reactspectrum.blob.core.windows.net/reactspectrum/0182ad0748bcdddf7eb010540c453f9a35a7c753/storybook/index.html?path=/story/drag-and-drop--nested-drop-regions&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom).
2. Move keyboard focus to the draggable element labelled "Drag me".
3. Press Enter key to start drag and drop operation.
4. Focus should move to the parent of the two nested drop targets.
5. Press Tab key to move focus to the child drop target.

## 🤔 Expected Behavior

6. Focus should move to the child drop target so that a user can complete the drag and drop operation using the keyboard alone.

## 😯 Current Behavior

6. Focus does not move to the child drop target, and a user cannot drop an item on the child drop target using the keyboard alone.

## 💁 Possible Solution

Problem is with how we determine the `dropTarget` within the `DragManager` `onFocus` event handler. We need to prioritize the `event.target` matching a `validDropTarget` over a `validDropTarget` containing the `event.target`, otherwise the parent element may match first.

Here is a suggested fix, change this:
```js
let dropTarget = this.validDropTargets.find(target => target.element.contains(e.target as HTMLElement));
```
to this:
```js
let dropTarget =
      this.validDropTargets.find(target => target.element === e.target as HTMLElement) ||
      this.validDropTargets.find(target => target.element.contains(e.target as HTMLElement));
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |  @react-aria/dnd@3.0.0
| Browser          | Chrome Version 108.0.5359.94 (Official Build) (arm64)
| Operating System | macOS 13.0.1 (22A400)

## 🧢 Your Company/Team

Adobe/Accessibility

### PR DESCRIPTION
**Pull request by majornista**
Problem is with how we determine the dropTarget within the DragManager onFocus event handler. We need to prioritize the event.target matching a validDropTarget over a validDropTarget containing the event.target, otherwise the parent element may match first.

Closes #3842 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 3842](https://github.com/adobe/react-spectrum/issue/3842).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/c42fb35488449eeae9fcb38139f02336ad9bc550/storybook/index.html?path=/story/drag-and-drop--nested-drop-regions&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom
2. Move keyboard focus to the draggable element labelled "Drag me".
3. Press Enter key to start drag and drop operation.
4. Focus should move to the parent of the two nested drop targets.
5. Press Tab key to move focus to the child drop target.
6. Focus should move to the child drop target so that a user can complete the drag and drop operation using the keyboard alone.
7. Press Enter to complete the drag and drop operation.

## 🧢 Your Project:

Adobe/Accessibility

### CODE REVIEW

**Review by reidbarber — APPROVED** [review]
LGTM

File: packages/@react-aria/dnd/test/dnd.test.js [hunk 1]
```diff
@@ -1992,6 +1992,143 @@ describe('useDrag and useDrop', function () {
...
+      });
+
+      act(() => jest.runAllTimers());
+      expect(document.activeElement).toBe(droppable2);
+      expect(droppable2).toHaveAttribute('aria-describedby');
+      expect(document.getElementById(droppable2.getAttribute('aria-describedby'))).toHaveTextContent('Press Enter to drop');
+      expect(announce).toHaveBeenCalledWith('Started dragging. Press Tab to navigate to a drop target, then press Enter to drop, or press Escape to cancel.');
+
+      expect(onDropEnter2).toHaveBeenCalledTimes(1);
+      expect(onDropEnter).toHaveBeenCalledTimes(0);
```
Comment (LFDanLu) [inline packages/@react-aria/dnd/test/dnd.test.js:2049]: Strange, this means the test is showing that the nested drop region got focus first right?

Comment (majornista) [reply packages/@react-aria/dnd/test/dnd.test.js:2049]: Yeah, I seem to recall  the test behaving in reverse order, but the storybook story behaving as expected. It was a while ago.

**Review by LFDanLu — APPROVED** [review]

**Review by devongovett — APPROVED** [review]
We will figure out the test later, and maybe look into sorting the valid drop targets so parents always come before children regardless of which order they are registered in. #2071

**Pull request merged**